modules/tensor_mechanics/src/materials/crystal_plasticity/CrystalPlasticityKalidindiBackstress.C

```diff
@@ -0,0 +1,207 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+/*
+   Constitutive model form:
+   Reference: Kalidindi, S. R., Bronkhorst, C. A., & Anand, L. (1992). Crystallographic texture evolution in bulk deformation processing of FCC metals. Journal of the Mechanics and Physics of Solids, 40(3), 537-569.
+
+   Plus an Armstrong-Frederick backstress term:
+   Reference: Armstrong, P. J., & Frederick, C. O. (1966). A mathematical representation of the multiaxial Bauschinger effect (Vol. 731). Berkeley, CA: Berkeley Nuclear Laboratories.
+
+   The flow rule of the shearing slip rate:
+   \dot{\gamma}^\alpha = \dot{\gamma}_o * |(\tau^\alpha - \chi^\alpha) / g^\alpha|^M * sgn(\tau^\alpha - \chi^\alpha)
+
+   Armstrong-Frederick backstress term:
+   \dot{\chi}^\alpha = h * \dot{\gamma}^\alpha - r * |\dot{\gamma}^\alpha| * \chi^\alpha
+
+   Reference: https://github.com/ngrilli/c_pfor_am/blob/main/src/materials/CrystalPlasticityDislocationUpdate.C
+*/
+
+#include "CrystalPlasticityKalidindiBackstress.h"
+
+registerMooseObject("TensorMechanicsApp", CrystalPlasticityKalidindiBackstress);
+ 
+InputParameters
+CrystalPlasticityKalidindiBackstress::validParams()
+{
+  InputParameters params = CrystalPlasticityKalidindiUpdate::validParams();
+  params.addClassDescription("Kalidindi version of homogeneous crystal plasticity considering backstress term.");
+  params.addParam<Real>("h",0.0,"Direct hardening coefficient for backstress");
+  params.addParam<Real>("h_D",0.0,"Dynamic recovery coefficient for backstress");
+
+  return params;
+}
+
+CrystalPlasticityKalidindiBackstress::CrystalPlasticityKalidindiBackstress(
+    const InputParameters & parameters)
+  : CrystalPlasticityKalidindiUpdate(parameters),
+
+    // Backstress parameters
+    _h(getParam<Real>("h")),
+    _h_D(getParam<Real>("h_D")),
+
+    // Backstress variable
+    _backstress(declareProperty<std::vector<Real>>("backstress")),
+    _backstress_old(getMaterialPropertyOld<std::vector<Real>>("backstress")),
+    _backstress_increment(_number_slip_systems, 0.0),
```

Comment:
I recommend that you also make `_backstress_increment` a material property vector as well (Please add the `_base_name` variable as suggested for lines 50 and 51 above)

Reply:
resolved